lib/puma/configuration.rb

```diff
@@ -137,6 +137,8 @@ def initialize(user_options={}, default_options = {}, &block)
       @file_dsl    = DSL.new(@options.file_options, self)
       @default_dsl = DSL.new(@options.default_options, self)
 
+      default_options[:preload_app] = (default_options[:workers] > 1 && Puma::Plugin.new.workers_supported?)
```

Comment:
Come to think of it, this could break things for people doing the "don't preload but do USR2 restarts instead" path. 

I think we need to check to see if other settings indicating people are doing that are set, like `prune_bundler`. So if `prune_bundler` is true, don't preload_app by _default_. But I'm not sure if we can detect it otherwise? 

+ Also let's find a way to check if we can do workers without calling into `Puma::Plugin`, that looks ugly.

Reply:
This breakage may be what the tests are showing you.

Reply:
How do we feel about this?

```			
# 1. workers_supported - #Puma::Plugin.new.workers_supported?)
workers_supported = !(Puma.jruby? || Puma.windows?)

# 2. preload app
if !@options[:prune_bundler]
  default_options[:preload_app] =(@options[:workers] > 1) && workers_supported 
end
```

Reply:
LGTM!